XMLHttpRequest/getallresponseheaders-cl.htm

```diff
@@ -0,0 +1,27 @@
+<!doctype html>
+<title>Casing of known headers</title>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<script>
+test(() => {
+  const client = new XMLHttpRequest
+  client.open("GET", "resources/header-content-length.asis", false)
+  client.send()
+  assert_equals(client.getAllResponseHeaders(), "CONTENT-LENGTH: 0\r\n")
+})
+test(() => {
+  const client = new XMLHttpRequest
+  client.open("GET", "resources/echo-headers.py", false)
+  client.setRequestHeader("THIS-IS-A-TEST", "1")
+  client.setRequestHeader("THIS-is-A-test", "2")
+  client.setRequestHeader("content-TYPE", "x/x")
+  client.send()
+  assert_regexp_match(client.responseText, /content-TYPE/)
+  assert_regexp_match(client.responseText, /THIS-IS-A-TEST: 1,/)
+})
+promise_test(() => {
+  return fetch("resources/echo-headers.py", {headers: [["THIS-is-A-test", 1], ["THIS-IS-A-TEST", 2]] }).then(res => res.text()).then(body => {
+    assert_regexp_match(body, /THIS-is-A-test: 1/)
```

Comment:
Copypasta. It doesn't matter much either way. I ended up trimming it from "1, 2" since we haven't quite settled on the space there.